src/block_disagg/block_disagg_read.c

```diff
@@ -112,38 +114,60 @@ __block_disagg_read(WT_SESSION_IMPL *session, WT_BLOCK_DISAGG *block_disagg, WT_
           "of %" PRIu32
           " doesn't match expected checksum "
           "of %" PRIu32,
-          block_disagg->name, size, (uintmax_t)disagg_id, swap.checksum, checksum);
+          block_disagg->name, size, (uintmax_t)page_id, swap.checksum, checksum);
 
     if (!F_ISSET(session, WT_SESSION_QUIET_CORRUPT_FILE))
-        WT_IGNORE_RET(__wt_bm_corrupt_dump(session, buf, 0, (wt_off_t)disagg_id, size, checksum));
+        WT_IGNORE_RET(__wt_bm_corrupt_dump(session, current, 0, (wt_off_t)page_id, size, checksum));
 
     /* Panic if a checksum fails during an ordinary read. */
     F_SET(S2C(session), WT_CONN_DATA_CORRUPTION);
     if (F_ISSET(session, WT_SESSION_QUIET_CORRUPT_FILE))
-        return (WT_ERROR);
-    WT_RET_PANIC(session, WT_ERROR, "%s: fatal read error", block_disagg->name);
+        WT_ERR(WT_ERROR);
+    WT_ERR_PANIC(session, WT_ERROR, "%s: fatal read error", block_disagg->name);
+err:
+done:
+    return (ret);
 }
 
 /*
  * __wt_block_disagg_read --
- *     Map or read address cookie referenced block into a buffer.
+ *     A basic read of a single block is not supported in disaggregated storage.
  */
 int
 __wt_block_disagg_read(WT_BM *bm, WT_SESSION_IMPL *session, WT_ITEM *buf,
   WT_PAGE_BLOCK_META *block_meta, const uint8_t *addr, size_t addr_size)
+{
```

Comment:
What is the difference of read multiple and read?

Reply:
Read multiple is supposed to return the last full image + all the subsequent deltas in one call. (This requires a different interface, so hence it is a different function.)

Reply:
Right, in the disagg block manager, the read() function will never be called, only read_multiple().